mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/EntityHelperGetMethodGenerator.java

```diff
@@ -18,37 +18,98 @@
 import com.datastax.oss.driver.api.core.data.GettableByName;
 import com.datastax.oss.driver.internal.mapper.processor.PartialClassGenerator;
 import com.datastax.oss.driver.internal.mapper.processor.ProcessorContext;
+import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableMap;
 import com.squareup.javapoet.ClassName;
 import com.squareup.javapoet.MethodSpec;
 import com.squareup.javapoet.ParameterSpec;
+import com.squareup.javapoet.TypeName;
 import com.squareup.javapoet.TypeSpec;
+import java.util.Map;
 import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
 
 public class EntityHelperGetMethodGenerator implements PartialClassGenerator {
 
   private final EntityDefinition entityDefinition;
 
+  private static final Map<TypeName, String> PRIMITIVE_ACCESSORS =
+      ImmutableMap.<TypeName, String>builder()
+          .put(TypeName.BOOLEAN, "Boolean")
+          .put(TypeName.BYTE, "Byte")
+          .put(TypeName.DOUBLE, "Double")
+          .put(TypeName.FLOAT, "Float")
+          .put(TypeName.INT, "Int")
+          .put(TypeName.LONG, "Long")
+          .build();
+
+  private final EntityHelperGenerator enclosingClass;
+
   public EntityHelperGetMethodGenerator(
       EntityDefinition entityDefinition,
       EntityHelperGenerator enclosingClass,
       ProcessorContext context) {
     this.entityDefinition = entityDefinition;
+    this.enclosingClass = enclosingClass;
   }
 
   @Override
   public void addMembers(TypeSpec.Builder classBuilder) {
 
-    MethodSpec.Builder extractBuilder =
+    MethodSpec.Builder getBuilder =
         MethodSpec.methodBuilder("get")
             .addAnnotation(Override.class)
             .addModifiers(Modifier.PUBLIC)
             .addParameter(
                 ParameterSpec.builder(ClassName.get(GettableByName.class), "source").build())
-            .returns(entityDefinition.getClassName())
-            // TODO generate proper implementation (JAVA-2112)
-            .addStatement("throw new $T($S)", UnsupportedOperationException.class, "TODO");
+            .returns(entityDefinition.getClassName());
+
+    TypeName returnType = entityDefinition.getClassName();
+    String returnName = "returnValue";
+    getBuilder.addStatement("$T $L = new $T()", returnType, returnName, returnType);
```

Comment:
Nit: you can number placeholders to avoid repeating the values:
```suggestion
    getBuilder.addStatement("$1T $2L = new $1T()", returnType, returnName);
```